qcodes/instrument/channel.py

```diff
@@ -529,3 +532,183 @@ def validate(self, value, context: str='') -> None:
             raise ValueError(
                 '{} is not part of the expected channel list; {}'.format(
                     repr(value), context))
+
+
+class AutoLoadableInstrumentChannel(InstrumentChannel):
+    """
+    This subclass provides extensions to auto-load channels
+    from instruments and adds methods to create and delete
+    channels when possible. Please note that `channel` in this
+    context does not necessarily mean a physical instrument channel,
+    but rather an instrument sub-module. For some instruments,
+    these sub-modules can be created and deleted at will.
+    """
+
+    @classmethod
+    def load_from_instrument(
+            cls, parent: Instrument, **kwargs) ->List['AutoLoadableInstrumentChannel']:
+        """
+        Discover and load channels from the instrument
```

Comment:
Mentioning `discover` here IMHO makes it a bit confusing.
```suggestion
        Load channels that already exist on the instrument
```